### ISSUE TITLE
Update site and product build maintenance

### ISSUE THREAD

**Issue by wimjongman** [Sloeber/arduino-eclipse-plugin#1027]
We will use this issue to track the cleanup of the product build and update site.

**Comment by wimjongman** [comment]
* The update site now only contains Sloeber
* Do not add features to the product files with the product editor. It will remove installmode=root
* The formatting options and warning settings should be defined by us per project.
* The .settings directory should be checked in.

* I have removed the application plugin.
** This removed the intro page. This has to be added again but now better because it was ugly as fork

**Comment by jantje** [comment]
The product has to be in the update site one way or another to be able to upgrade the product from lets say 4.3 to 4.4
At least to fix a problem like that I added the product and that fixed it.
Note that due to the deletion of the application plugin some bitmaps are no longer available (like the icon used in the toolbar)

**Comment by jantje** [comment]
And adding the very latests sloeber to eclipse-java-2018-09-win32-x86_64 produces this error on my system
```
Cannot complete the install because one or more required items could not be found.
  Software being installed: Sloeber, the Eclipse Arduino IDE (plugin version) 4.3.1.201810032102 (io.sloeber.feature.feature.group 4.3.1.201810032102)
  Missing requirement: Sloeber user interface 4.3.1.201810032102 (io.sloeber.ui 4.3.1.201810032102) requires 'java.package; org.eclipse.nebula.widgets.oscilloscope.multichannel 0.0.0' but it could not be found
  Cannot satisfy dependency:
    From: Sloeber, the Eclipse Arduino IDE (plugin version) 4.3.1.201810032102 (io.sloeber.feature.feature.group 4.3.1.201810032102)
    To: org.eclipse.equinox.p2.iu; io.sloeber.ui [4.3.1.201810032102,4.3.1.201810032102]
```

**Comment by wimjongman** [comment]
> 
> 
> And adding the very latests sloeber to eclipse-java-2018-09-win32-x86_64 produces this error on my system
> 
> ```
> Cannot complete the install because one or more required items could not be found.
>   Software being installed: Sloeber, the Eclipse Arduino IDE (plugin version) 4.3.1.201810032102 (io.sloeber.feature.feature.group 4.3.1.201810032102)
>   Missing requirement: Sloeber user interface 4.3.1.201810032102 (io.sloeber.ui 4.3.1.201810032102) requires 'java.package; org.eclipse.nebula.widgets.oscilloscope.multichannel 0.0.0' but it could not be found
>   Cannot satisfy dependency:
>     From: Sloeber, the Eclipse Arduino IDE (plugin version) 4.3.1.201810032102 (io.sloeber.feature.feature.group 4.3.1.201810032102)
>     To: org.eclipse.equinox.p2.iu; io.sloeber.ui [4.3.1.201810032102,4.3.1.201810032102]
> ```

Thanks I will take a look at that. I did not try a virgin Eclipse product,

**Comment by jantje** [comment]
I also found that the perspective and intro were in the application.
Do you have a link to a document how you can easily get parts of deleted stuff via git?

**Comment by wimjongman** [comment]
**I already restored the application this morning.**

> I also found that the perspective and intro were in the application.
> Do you have a link to a document how you can easily get parts of deleted stuff via git?

In the history, click on the commit just before the delete and then create new branch.

**Comment by wimjongman** [comment]
Oopsie, I forgot to push that part. Sorry dude. I hope you did not work on it.

**Comment by jantje** [comment]
I'm still on the build issue :-(
The git security issue is fixed and jenkins runs on a safe tmp folder but tycho-p2-repository-plugin seems to get the tmp folder from somewhere else

**Comment by jantje** [comment]
I see you upgraded the tycho version.
I downgraded it to test whether this is the root cause of the /tmp issue on my build server

**Comment by jantje** [comment]
reverting to the old tycho fixed the issue.
Nice to know the root cause
From what I investigated this is caused by linux not having a standard way to set a  temp variable so java provides /tmp
I need to set the  -Djava.io.tmpdir=/to/something/that/can/handle/big/files

**Comment by jantje** [comment]
Och yes I tested the upgrade and that one work :-) Congratulations.
The update site only contains nebula and gnumcu stuff :-) a huge improvement

**Comment by wimjongman** [comment]
Yes, I had to include those because the update sites in the category.xml are not used so it seems.

**Comment by jantje** [comment]
And how is the cdt stuff found?  I mean which update sites are used?

**Comment by wimjongman** [comment]
That is in the o.e.d/release/latest repository.

**Comment by jantje** [comment]
Further investigating the "out of disk space" problem on the build server confirms that the tycho upgrade causes this to happen.
I have been investigating the startup scripts of my tomcat and I think I correctly set the java default temp folder to a safe place but I keep getting a full /tmp folder :-(
This is the command that starts tomcat
`"/usr/local/jre/bin/java" "-Djava.util.logging.config.file=/share/CACHEDEV1_DATA/.qpkg/Tomcat8/tomcat8/conf/logging.properties" -Djava.util.logging.manager=org.apache.juli.ClassLoaderLogManager -Djdk.tls.ephemeralDHKeySize=2048 -Djava.protocol.handler.pkgs=org.apache.catalina.webresources -Djava.io.tmpdir=/share/homes/admin/tmp/ -Dignore.endorsed.dirs="" -classpath "/share/CACHEDEV1_DATA/.qpkg/Tomcat8/tomcat8/bin/bootstrap.jar:/share/CACHEDEV1_DATA/.qpkg/Tomcat8/tomcat8/bin/tomcat-juli.jar" -Dcatalina.base="/share/CACHEDEV1_DATA/.qpkg/Tomcat8/tomcat8" -Dcatalina.home="/share/CACHEDEV1_DATA/.qpkg/Tomcat8/tomcat8" -Djava.io.tmpdir="/share/homes/admin/tmp" org.apache.catalina.startup.Bootstrap start`
 -Djava.io.tmpdir="/share/homes/admin/tmp" is set 2 times because I had set  
`export CATALINA_TMPDIR=/share/homes/admin/tmp`
and as this didn't help I also set
`export CATALINA_OPTS="-Djava.io.tmpdir=/share/homes/admin/tmp/"`
and when investigating further I that the execute command contained
`-Djava.io.tmpdir="\"$CATALINA_TMPDIR\"" `

**Comment by jantje** [comment]
I created a bug https://bugs.eclipse.org/bugs/show_bug.cgi?id=539820
I'm reopening as I revert to V0.24

**Comment by wimjongman** [comment]
V0.24 will not build on java 9

**Comment by jantje** [comment]
I agree this needs to be fixed but...
I have been focussing 3 days now on getting this to work.
I need a break and for now it is working.
I hope time will bring ideas/solutions/energy

**Comment by jantje** [comment]
I thought I reported success, seems I was to tired
It works with adding the MAVEN_OPTS="-Djava.io.tmpdir=/somewhere" environment variable.
I moved to tycho V 1.2 (as I assume that is the latest and greatest) but it was 1.1 so in case it needs to be 1.1 feel free to change.
